recipes/imagemagick/cmake/conanfile.py

```diff
@@ -0,0 +1,445 @@
+        tc.cache_variables["X11_DELEGATE"] = self.options.get_safe("with_x11", False)
+        tc.cache_variables["XML_DELEGATE"] = self.options.with_xml2
+        tc.cache_variables["ZLIB_DELEGATE"] = self.options.with_zlib
+        tc.cache_variables["ZSTD_DELEGATE"] = self.options.with_zstd
+        tc.generate()
+
+        # TODO: fix this bug in the libheif recipe
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.dependencies["libheif"].cpp_info.components["heif"].system_libs.append("dl")
+
+        cmake_names = {
+            "autotrace": "AUTOTRACE",
+            "bzip2": "BZip2",
+            "cairo": "Cairo",
+            "djvu": "DJVU",
+            "dps": "DPS",
+            "fftw": "FFTW",
+            "flashpix": "FlashPIX",
+            "flif": "FLIF",
+            "fontconfig": "Fontconfig",
+            "freetype": "Freetype",
+            "ghostscript": "Ghostscript",
+            "gvc": "GVC",
+            "jbig": "JBIG",
+            "jemalloc": "Jemalloc",
+            "jpeg": "JPEG",
+            "lcms": "lcms",
+            "libheif": "libheif",
+            "libjxl": "libjxl",
+            "libpng": "PNG",
+            "libraw": "libraw",
+            "libtiff": "TIFF",
+            "libwebp": "WebP",
+            "libxml2": "LibXml2",
+            "lqr": "Lqr",
+            "ltdl": "LTDL",
+            "opencl-headers": "OpenCL",
+            "openexr": "OpenEXR",
+            "openjpeg": "OpenJPEG",
+            "pango": "Pango",
+            "raqm": "RAQM",
+            "rsvg": "Rsvg",
+            "umem": "UMEM",
+            "webpmux": "WEBPMUX",
+            "wmf": "WMF",
+            "xz_utils": "LibLZMA",
+            "zlib": "ZLIB",
+            "zstd": "zstd",
+        }
+        deps = CMakeDeps(self)
+        for dep, cmake_name in cmake_names.items():
+            deps.set_property(dep, "cmake_file_name", cmake_name)
+            deps.set_property(dep, "cmake_target_name", f"{cmake_name}::{cmake_name}")
+        deps.generate()
+
+        deps = PkgConfigDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        # Fix mismatch with Conan's CMake variable capitalization.
+        # Also add REQUIRED since we explicitly enable/disable all dependencies.
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "    find_package(${MAGICK_FIND_NAME} QUIET)\n",
+                        "    find_package(${MAGICK_FIND_NAME} REQUIRED)\n"
+                        "    string(TOUPPER ${MAGICK_FIND_NAME} name_upper)\n"
+                        "    set(${name_upper}_FOUND ${${MAGICK_FIND_NAME}_FOUND})\n")
+        # Fix incorrect .so install location
+        for subdir in ["MagickCore", "MagickWand", "Magick++"]:
+            replace_in_file(self, os.path.join(self.source_folder, subdir, "CMakeLists.txt"),
+                            "LIBRARY DESTINATION bin\n",
+                            "LIBRARY DESTINATION lib\n")
```

Comment:
@Cyriuz Could you fix this minor issue upstream in you repository, please? Only `RUNTIME` libraries are normally installed under `bin`.

Amazing work on the CMake build system, by the way. Thank you so much!

Reply:
Will take a look! Thank you for continuing on this!

Reply:
I did it! Just a heads up, the latest version I have updated the cmake for is 7.1.1-4, I can see that there's a few changes made in the makefiles between that and 7.1.1-22. It's possible there's tweaks that has to be made for it to be building everything correctly. I don't have time to fix it this week probably so either if you want to take a look and make PRs I'll look at them asap or if we want to be safer lock it to 7.1.1-4 to begin with?

Reply:
Opened a PR: https://github.com/Cyriuz/ImageMagick/pull/1